File: frontend/console/src/api/signal-acquisition.js
import Axios from "@/http";

export function createSignalAcquisition(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-acquisition', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveSignalAcquisition(signalAcquisitionId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/signal-acquisition/${ signalAcquisitionId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateSignalAcquisition(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/signal-acquisition', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalAcquisition(signalAcquisitionId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/signal-acquisition/${ signalAcquisitionId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalAcquisitionWithFlag(signalAcquisitionId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/signal-acquisition/${ signalAcquisitionId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllSignalAcquisitions() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/signal-acquisition/all').then(res => {
            resolve(res)
        })
    })
}

export function querySignalAcquisition (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-acquisition/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/signal_acquisition.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_SIGNAL_ACQUISITION = "/config/signal-acquisition"
const URL_ROUTER_SIGNAL_ACQUISITION_BY_ID = "/config/signal-acquisition/:signal-acquisition-id"
const URL_ROUTER_SIGNAL_ACQUISITION_QUERY = "/config/signal-acquisition/query"
const URL_ROUTER_SIGNAL_ACQUISITION_ALL = "/config/signal-acquisition/all"

func SignalAcquisitionRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_SIGNAL_ACQUISITION, config.CreateSignalAcquisition)
	r.GET(URL_ROUTER_SIGNAL_ACQUISITION_BY_ID, config.RetrieveSignalAcquisition)
	r.PUT(URL_ROUTER_SIGNAL_ACQUISITION, config.UpdateSignalAcquisition)
	r.DELETE(URL_ROUTER_SIGNAL_ACQUISITION_BY_ID, config.DeleteSignalAcquisition)
	r.POST(URL_ROUTER_SIGNAL_ACQUISITION_QUERY, config.QuerySignalAcquisition)
	r.GET(URL_ROUTER_SIGNAL_ACQUISITION_ALL, config.GetAllSignalAcquisitions)
	return r
}